kubecontrolplane/v1/types.go

```diff
@@ -0,0 +1,187 @@
+
+	// authConfig configures authentication options in addition to the standard
+	// oauth token and client certificate authenticators
+	AuthConfig MasterAuthConfig `json:"authConfig" protobuf:"bytes,2,opt,name=authConfig"`
+
+	// aggregatorConfig has options for configuring the aggregator component of the API server.
+	AggregatorConfig AggregatorConfig `json:"aggregatorConfig" protobuf:"bytes,3,opt,name=aggregatorConfig"`
+
+	// kubeletClientInfo contains information about how to connect to kubelets
+	KubeletClientInfo KubeletConnectionInfo `json:"kubeletClientInfo" protobuf:"bytes,4,opt,name=kubeletClientInfo"`
+
+	// servicesSubnet is the subnet to use for assigning service IPs
+	ServicesSubnet string `json:"servicesSubnet" protobuf:"bytes,5,opt,name=servicesSubnet"`
+	// servicesNodePortRange is the range to use for assigning service public ports on a host.
+	ServicesNodePortRange string `json:"servicesNodePortRange" protobuf:"bytes,6,opt,name=servicesNodePortRange"`
+
+	// legacyServiceServingCertSignerCABundle is the old service serving cert signer before we switched to a separate controller
+	// TODO this should be removable in a later release after we've completed migration
+	LegacyServiceServingCertSignerCABundle string `json:"legacyServiceServingCertSignerCABundle" protobuf:"bytes,7,opt,name=legacyServiceServingCertSignerCABundle"`
+
+	// UserAgentMatchingConfig controls how API calls from *voluntarily* identifying clients will be handled.  THIS DOES NOT DEFEND AGAINST MALICIOUS CLIENTS!
+	// TODO I think we should just drop this feature.
+	UserAgentMatchingConfig UserAgentMatchingConfig `json:"userAgentMatchingConfig" protobuf:"bytes,8,opt,name=userAgentMatchingConfig"`
+
+	// imagePolicyConfig feeds the image policy admission plugin
+	// TODO make it an admission plugin config
+	ImagePolicyConfig KubeAPIServerImagePolicyConfig `json:"imagePolicyConfig" protobuf:"bytes,9,opt,name=imagePolicyConfig"`
+
+	// projectConfig feeds an admission plugin
+	// TODO make it an admission plugin config
+	ProjectConfig KubeAPIServerProjectConfig `json:"projectConfig" protobuf:"bytes,10,opt,name=projectConfig"`
+
+	// serviceAccountPublicKeyFiles is a list of files, each containing a PEM-encoded public RSA key.
+	// (If any file contains a private key, the public portion of the key is used)
+	// The list of public keys is used to verify presented service account tokens.
+	// Each key is tried in order until the list is exhausted or verification succeeds.
+	// If no keys are specified, no service account authentication will be available.
+	ServiceAccountPublicKeyFiles []string `json:"serviceAccountPublicKeyFiles" protobuf:"bytes,11,rep,name=serviceAccountPublicKeyFiles"`
+
+	// oauthConfig, if present start the /oauth endpoint in this process
+	OAuthConfig *osinv1.OAuthConfig `json:"oauthConfig" protobuf:"bytes,13,opt,name=oauthConfig"`
+
+	// TODO this needs to be removed.
```

Comment:
can we remove this now? once we claim this 'v1', removing it won't be that easy I think

Reply:
> can we remove this now? once we claim this 'v1', removing it won't be that easy I think

Not without effectively blocking operator progress